src/PHPUnit/Constraint/IsCurrentModuleNameConstraint.php

```diff
@@ -39,7 +40,10 @@ public function getCurrentModuleName(): string
         $applicationConfig = $this->getTestCase()->getApplicationConfig();
 
         // Find Module from Controller
-        foreach ($applicationConfig['modules'] as $appModules) {
+        foreach ($applicationConfig['modules'] as $moduleName => $appModules) {
+            if (is_object($appModules)) {
```

Comment:
Why would that ever be an `object`? This is the config, not the runtime list of modules?

Reply:
The ModuleManager.php also allows you to load objects associated with the name of the module.

Config without object
```php
'modules' => [
    'Laminas\Filter',
    'Laminas\Form',
    'Laminas\I18n',
    'Laminas\Router',
   // ...
],
```

Config with objects
```php
'modules' => [
    'Laminas\Filter' => new Laminas\Filter\Module(),
    'Laminas\Form'   => new Laminas\Form\Module(),
    'Laminas\I18n'   => new Laminas\I18n\Module(),
    'Laminas\Router' => new Laminas\Router\Module(),
   // ...
],
```

https://github.com/laminas/laminas-modulemanager/blob/2.11.x/src/ModuleManager.php
```php
    /**
     * Handle the loadModules event
     *
     * @return void
     */
    public function onLoadModules()
    {
        if (true === $this->modulesAreLoaded) {
            return;
        }

        foreach ($this->getModules() as $moduleName => $module) {
            if (is_object($module)) {
                if (! is_string($moduleName)) {
                    throw new Exception\RuntimeException(sprintf(
                        'Module (%s) must have a key identifier.',
                        get_class($module)
                    ));
                }
                $module = [$moduleName => $module];
            }

            $this->loadModule($module);
        }

        $this->modulesAreLoaded = true;
    }
```

The idea was to keep this behavior in the unit test.

Reply:
Wondering why/whether this is valid though:

```php
'modules' => [
    'Laminas\Filter' => new Laminas\Filter\Module(),
    'Laminas\Form'   => new Laminas\Form\Module(),
    'Laminas\I18n'   => new Laminas\I18n\Module(),
    'Laminas\Router' => new Laminas\Router\Module(),
   // ...
],
```

It seems really sketchy to me :-\

Reply:
It seemed to me that this behavior optimizes the loading of the application.